pkg/proxyserver/operator/eventhandler/triggerallclustermanagementaddon_event.go

```diff
@@ -0,0 +1,51 @@
+package eventhandler
+
+import (
+	"context"
+
+	"k8s.io/client-go/util/workqueue"
+	addonv1alpha1 "open-cluster-management.io/api/addon/v1alpha1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+var _ handler.EventHandler = &TriggerAllClusterManagementAddonsHandler{}
+
+type TriggerAllClusterManagementAddonsHandler struct {
+	client.Client
+}
+
+func (m TriggerAllClusterManagementAddonsHandler) Create(event event.CreateEvent, limitingInterface workqueue.RateLimitingInterface) {
+	m.findClusterProxyAddon(limitingInterface)
+}
+
+func (m TriggerAllClusterManagementAddonsHandler) Update(event event.UpdateEvent, limitingInterface workqueue.RateLimitingInterface) {
+	m.findClusterProxyAddon(limitingInterface)
+}
+
+func (m TriggerAllClusterManagementAddonsHandler) Delete(event event.DeleteEvent, limitingInterface workqueue.RateLimitingInterface) {
+	m.findClusterProxyAddon(limitingInterface)
+}
+
+func (m TriggerAllClusterManagementAddonsHandler) Generic(event event.GenericEvent, limitingInterface workqueue.RateLimitingInterface) {
+	m.findClusterProxyAddon(limitingInterface)
+}
+
+// findClusterProxyAddon will triger clustermanagementaddon on all managed clusters to reconcile.
+func (m *TriggerAllClusterManagementAddonsHandler) findClusterProxyAddon(limitingInterface workqueue.RateLimitingInterface) {
+	list := &addonv1alpha1.ClusterManagementAddOnList{}
+	err := m.Client.List(context.TODO(), list)
+	if err != nil {
+		return
+	}
+	for _, addon := range list.Items {
+		if addon.Spec.AddOnConfiguration.CRDName == crdName {
```

Comment:
this is being deperecated. I think when Resolver is changed, you will always need to update the certain addon, right? it does not need to check the crd name. for cluster set, it is a bit different. You would also need to check if the clusterset is in a certain resolver.

Reply:
A little confused here, "it does not need to check the crd name."
If we don't check the CRD name, is there any way else for us to know if it's a "cluster-proxy" addon?

Reply:
Split a event handler for `clusterset` having match check before insert into queue.

Reply:
you can use the name of the addon.

Reply:
Do we mean this field? https://github.com/open-cluster-management-io/cluster-proxy/blob/641a5dbc71eed83423abb36960c931972f1ce9c7/charts/cluster-proxy/templates/clustermanagementaddon.yaml#L4

Is there any possibility that a customer only use our image and a customized chart with this field doesn't equals to "cluster-proxy"?

Reply:
will it work in addon-framework? I think the addon-name is set in addon-framework

Reply:
Got it. Also thanks to @skeeey for the explanation. 
Updated all filter functions to using the name instead of CRDName.